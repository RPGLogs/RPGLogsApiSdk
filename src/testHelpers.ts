import { buildSdk } from './main';
import btoa from 'btoa';

export function buildTestSdk(): ReturnType<typeof buildSdk> {
  return buildSdk(<string>process.env.TEST_ACCESS_TOKEN);
}

export async function getAccessToken(
  clientId: string,
  clientSecret: string,
  game = 'warcraft'
): Promise<string> {
  if (game === 'warcraft') game = 'www.warcraft';

  const authHeader = 'Basic ' + btoa(clientId + ':' + clientSecret);

  const response = await fetch(`https://${game}logs.com/oauth/token`, {
    method: 'POST',
    headers: {
      Authorization: authHeader,
      'Content-Type': 'application/x-www-form-urlencoded',
    },
    body: 'grant_type=client_credentials',
  });

  const json = await response.json();

  if (response.status === 200) {
    return json.access_token;
  } else {
    throw new Error('Response was not OK: ' + JSON.stringify(json ?? {}));
  }
}
