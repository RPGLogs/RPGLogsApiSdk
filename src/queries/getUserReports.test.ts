import { buildTestSdk } from '../testHelpers';

test('returns data', async () => {
  const response = await buildTestSdk().getUserReports({
    userId: 381152,
    limit: 1,
    zoneId: 24,
    includeFights: true,
  });

  expect(response?.reportData?.reports).toMatchSnapshot();
});
