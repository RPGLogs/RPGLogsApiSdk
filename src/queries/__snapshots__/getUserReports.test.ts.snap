// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns data 1`] = `
Object {
  "current_page": 1,
  "data": Array [
    Object {
      "code": "bL2dKaz3ZjDgh1vQ",
      "endTime": 1602170539199,
      "exportedSegments": 49,
      "fights": Array [
        Object {
          "averageItemLevel": 462,
          "bossPercentage": 0,
          "completeRaid": false,
          "difficulty": 4,
          "encounterID": 2329,
          "endTime": 314593,
          "fightPercentage": 0,
          "id": 1,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": true,
          "lastPhase": 1,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "Wrathion",
          "rating": null,
          "size": 30,
          "startTime": 188780,
          "wipeCalledTime": null,
        },
        Object {
          "averageItemLevel": null,
          "bossPercentage": null,
          "completeRaid": false,
          "difficulty": null,
          "encounterID": 0,
          "endTime": 424015,
          "fightPercentage": null,
          "id": 2,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": null,
          "lastPhase": 0,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "Monstrous Behemoth",
          "rating": null,
          "size": null,
          "startTime": 340169,
          "wipeCalledTime": null,
        },
        Object {
          "averageItemLevel": null,
          "bossPercentage": null,
          "completeRaid": false,
          "difficulty": null,
          "encounterID": 0,
          "endTime": 550667,
          "fightPercentage": null,
          "id": 3,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": null,
          "lastPhase": 0,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "Anubisath Sentinel",
          "rating": null,
          "size": null,
          "startTime": 454312,
          "wipeCalledTime": null,
        },
        Object {
          "averageItemLevel": 462.76666259765625,
          "bossPercentage": 0,
          "completeRaid": false,
          "difficulty": 4,
          "encounterID": 2334,
          "endTime": 802118,
          "fightPercentage": 0,
          "id": 4,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": true,
          "lastPhase": 0,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "Prophet Skitra",
          "rating": null,
          "size": 30,
          "startTime": 605359,
          "wipeCalledTime": null,
        },
        Object {
          "averageItemLevel": null,
          "bossPercentage": null,
          "completeRaid": false,
          "difficulty": null,
          "encounterID": 0,
          "endTime": 805775,
          "fightPercentage": null,
          "id": 5,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": null,
          "lastPhase": 0,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "Unknown",
          "rating": null,
          "size": null,
          "startTime": 805775,
          "wipeCalledTime": null,
        },
        Object {
          "averageItemLevel": null,
          "bossPercentage": null,
          "completeRaid": false,
          "difficulty": null,
          "encounterID": 0,
          "endTime": 890275,
          "fightPercentage": null,
          "id": 6,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": null,
          "lastPhase": 0,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "Willing Sacrifice",
          "rating": null,
          "size": null,
          "startTime": 843755,
          "wipeCalledTime": null,
        },
        Object {
          "averageItemLevel": null,
          "bossPercentage": null,
          "completeRaid": false,
          "difficulty": null,
          "encounterID": 0,
          "endTime": 950611,
          "fightPercentage": null,
          "id": 7,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": null,
          "lastPhase": 0,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "Monstrous Behemoth",
          "rating": null,
          "size": null,
          "startTime": 915472,
          "wipeCalledTime": null,
        },
        Object {
          "averageItemLevel": null,
          "bossPercentage": null,
          "completeRaid": false,
          "difficulty": null,
          "encounterID": 0,
          "endTime": 994639,
          "fightPercentage": null,
          "id": 8,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": null,
          "lastPhase": 0,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "Hepthys",
          "rating": null,
          "size": null,
          "startTime": 961757,
          "wipeCalledTime": null,
        },
        Object {
          "averageItemLevel": null,
          "bossPercentage": null,
          "completeRaid": false,
          "difficulty": null,
          "encounterID": 0,
          "endTime": 1090757,
          "fightPercentage": null,
          "id": 9,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": null,
          "lastPhase": 0,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "Aqir Shadowcrafter",
          "rating": null,
          "size": null,
          "startTime": 1012639,
          "wipeCalledTime": null,
        },
        Object {
          "averageItemLevel": null,
          "bossPercentage": null,
          "completeRaid": false,
          "difficulty": null,
          "encounterID": 0,
          "endTime": 1118394,
          "fightPercentage": null,
          "id": 10,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": null,
          "lastPhase": 0,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "Maut",
          "rating": null,
          "size": null,
          "startTime": 1108477,
          "wipeCalledTime": null,
        },
        Object {
          "averageItemLevel": 463.6666564941406,
          "bossPercentage": 0,
          "completeRaid": false,
          "difficulty": 4,
          "encounterID": 2327,
          "endTime": 1366454,
          "fightPercentage": 0,
          "id": 11,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": true,
          "lastPhase": 1,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "Maut",
          "rating": null,
          "size": 30,
          "startTime": 1201457,
          "wipeCalledTime": null,
        },
        Object {
          "averageItemLevel": null,
          "bossPercentage": null,
          "completeRaid": false,
          "difficulty": null,
          "encounterID": 0,
          "endTime": 1514277,
          "fightPercentage": null,
          "id": 12,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": null,
          "lastPhase": 0,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "Gryth'ax the Executioner",
          "rating": null,
          "size": null,
          "startTime": 1449625,
          "wipeCalledTime": null,
        },
        Object {
          "averageItemLevel": null,
          "bossPercentage": null,
          "completeRaid": false,
          "difficulty": null,
          "encounterID": 0,
          "endTime": 1884650,
          "fightPercentage": null,
          "id": 13,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": null,
          "lastPhase": 0,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "Jesh'ra",
          "rating": null,
          "size": null,
          "startTime": 1567575,
          "wipeCalledTime": null,
        },
        Object {
          "averageItemLevel": null,
          "bossPercentage": null,
          "completeRaid": false,
          "difficulty": null,
          "encounterID": 0,
          "endTime": 1907396,
          "fightPercentage": null,
          "id": 14,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": null,
          "lastPhase": 0,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "Dark Inquisitor Xanesh",
          "rating": null,
          "size": null,
          "startTime": 1901810,
          "wipeCalledTime": null,
        },
        Object {
          "averageItemLevel": null,
          "bossPercentage": null,
          "completeRaid": false,
          "difficulty": null,
          "encounterID": 0,
          "endTime": 1968934,
          "fightPercentage": null,
          "id": 15,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": null,
          "lastPhase": 0,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "Queen Azshara",
          "rating": null,
          "size": null,
          "startTime": 1968934,
          "wipeCalledTime": null,
        },
        Object {
          "averageItemLevel": 463.9666748046875,
          "bossPercentage": 0,
          "completeRaid": false,
          "difficulty": 4,
          "encounterID": 2328,
          "endTime": 2193455,
          "fightPercentage": 0,
          "id": 16,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": true,
          "lastPhase": 0,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "Dark Inquisitor Xanesh",
          "rating": null,
          "size": 30,
          "startTime": 1980356,
          "wipeCalledTime": null,
        },
        Object {
          "averageItemLevel": null,
          "bossPercentage": null,
          "completeRaid": false,
          "difficulty": null,
          "encounterID": 0,
          "endTime": 2197397,
          "fightPercentage": null,
          "id": 17,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": null,
          "lastPhase": 0,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "Queen Azshara",
          "rating": null,
          "size": null,
          "startTime": 2197397,
          "wipeCalledTime": null,
        },
        Object {
          "averageItemLevel": null,
          "bossPercentage": null,
          "completeRaid": false,
          "difficulty": null,
          "encounterID": 0,
          "endTime": 2357453,
          "fightPercentage": null,
          "id": 18,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": null,
          "lastPhase": 0,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "Vexiona",
          "rating": null,
          "size": null,
          "startTime": 2248056,
          "wipeCalledTime": null,
        },
        Object {
          "averageItemLevel": null,
          "bossPercentage": null,
          "completeRaid": false,
          "difficulty": null,
          "encounterID": 0,
          "endTime": 2380085,
          "fightPercentage": null,
          "id": 19,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": null,
          "lastPhase": 0,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "Vexiona",
          "rating": null,
          "size": null,
          "startTime": 2363461,
          "wipeCalledTime": null,
        },
        Object {
          "averageItemLevel": 464.20001220703125,
          "bossPercentage": 0,
          "completeRaid": false,
          "difficulty": 4,
          "encounterID": 2336,
          "endTime": 2803231,
          "fightPercentage": 0,
          "id": 20,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": true,
          "lastPhase": 1,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "Vexiona",
          "rating": null,
          "size": 30,
          "startTime": 2607599,
          "wipeCalledTime": null,
        },
        Object {
          "averageItemLevel": null,
          "bossPercentage": null,
          "completeRaid": false,
          "difficulty": null,
          "encounterID": 0,
          "endTime": 3013106,
          "fightPercentage": null,
          "id": 21,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": null,
          "lastPhase": 0,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "Deresh of the Nothingness",
          "rating": null,
          "size": null,
          "startTime": 2845949,
          "wipeCalledTime": null,
        },
        Object {
          "averageItemLevel": null,
          "bossPercentage": null,
          "completeRaid": false,
          "difficulty": null,
          "encounterID": 0,
          "endTime": 3046221,
          "fightPercentage": null,
          "id": 22,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": null,
          "lastPhase": 0,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "The Hivemind",
          "rating": null,
          "size": null,
          "startTime": 3040796,
          "wipeCalledTime": null,
        },
        Object {
          "averageItemLevel": 464.5,
          "bossPercentage": 0,
          "completeRaid": false,
          "difficulty": 4,
          "encounterID": 2333,
          "endTime": 3254865,
          "fightPercentage": 0,
          "id": 23,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": true,
          "lastPhase": 0,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "The Hivemind",
          "rating": null,
          "size": 30,
          "startTime": 3102455,
          "wipeCalledTime": null,
        },
        Object {
          "averageItemLevel": null,
          "bossPercentage": null,
          "completeRaid": false,
          "difficulty": null,
          "encounterID": 0,
          "endTime": 3279463,
          "fightPercentage": null,
          "id": 24,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": null,
          "lastPhase": 0,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "Engorged Maggot",
          "rating": null,
          "size": null,
          "startTime": 3277654,
          "wipeCalledTime": null,
        },
        Object {
          "averageItemLevel": null,
          "bossPercentage": null,
          "completeRaid": false,
          "difficulty": null,
          "encounterID": 0,
          "endTime": 3332443,
          "fightPercentage": null,
          "id": 25,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": null,
          "lastPhase": 0,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "Aqir Crusher",
          "rating": null,
          "size": null,
          "startTime": 3293572,
          "wipeCalledTime": null,
        },
        Object {
          "averageItemLevel": null,
          "bossPercentage": null,
          "completeRaid": false,
          "difficulty": null,
          "encounterID": 0,
          "endTime": 3409022,
          "fightPercentage": null,
          "id": 26,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": null,
          "lastPhase": 0,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "Abyssal Watcher",
          "rating": null,
          "size": null,
          "startTime": 3344822,
          "wipeCalledTime": null,
        },
        Object {
          "averageItemLevel": null,
          "bossPercentage": null,
          "completeRaid": false,
          "difficulty": null,
          "encounterID": 0,
          "endTime": 3421006,
          "fightPercentage": null,
          "id": 27,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": null,
          "lastPhase": 0,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "Ra-den the Despoiled",
          "rating": null,
          "size": null,
          "startTime": 3409044,
          "wipeCalledTime": null,
        },
        Object {
          "averageItemLevel": null,
          "bossPercentage": null,
          "completeRaid": false,
          "difficulty": null,
          "encounterID": 0,
          "endTime": 3435755,
          "fightPercentage": null,
          "id": 28,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": null,
          "lastPhase": 0,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "Abyssal Watcher",
          "rating": null,
          "size": null,
          "startTime": 3421044,
          "wipeCalledTime": null,
        },
        Object {
          "averageItemLevel": null,
          "bossPercentage": null,
          "completeRaid": false,
          "difficulty": null,
          "encounterID": 0,
          "endTime": 3470515,
          "fightPercentage": null,
          "id": 29,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": null,
          "lastPhase": 0,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "Ra-den the Despoiled",
          "rating": null,
          "size": null,
          "startTime": 3462229,
          "wipeCalledTime": null,
        },
        Object {
          "averageItemLevel": null,
          "bossPercentage": null,
          "completeRaid": false,
          "difficulty": null,
          "encounterID": 0,
          "endTime": 3513838,
          "fightPercentage": null,
          "id": 30,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": null,
          "lastPhase": 0,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "Ra-den the Despoiled",
          "rating": null,
          "size": null,
          "startTime": 3505493,
          "wipeCalledTime": null,
        },
        Object {
          "averageItemLevel": 464.9333190917969,
          "bossPercentage": 0,
          "completeRaid": false,
          "difficulty": 4,
          "encounterID": 2331,
          "endTime": 3805593,
          "fightPercentage": 0,
          "id": 31,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": true,
          "lastPhase": 2,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "Ra-den the Despoiled",
          "rating": null,
          "size": 30,
          "startTime": 3594247,
          "wipeCalledTime": null,
        },
        Object {
          "averageItemLevel": null,
          "bossPercentage": null,
          "completeRaid": false,
          "difficulty": null,
          "encounterID": 0,
          "endTime": 4085797,
          "fightPercentage": null,
          "id": 32,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": null,
          "lastPhase": 0,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "Drest'agath",
          "rating": null,
          "size": null,
          "startTime": 3867079,
          "wipeCalledTime": null,
        },
        Object {
          "averageItemLevel": null,
          "bossPercentage": null,
          "completeRaid": false,
          "difficulty": null,
          "encounterID": 0,
          "endTime": 4105665,
          "fightPercentage": null,
          "id": 33,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": null,
          "lastPhase": 0,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "Corrosive Digester",
          "rating": null,
          "size": null,
          "startTime": 4097750,
          "wipeCalledTime": null,
        },
        Object {
          "averageItemLevel": null,
          "bossPercentage": null,
          "completeRaid": false,
          "difficulty": null,
          "encounterID": 0,
          "endTime": 4159142,
          "fightPercentage": null,
          "id": 34,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": null,
          "lastPhase": 0,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "Corrosive Digester",
          "rating": null,
          "size": null,
          "startTime": 4118398,
          "wipeCalledTime": null,
        },
        Object {
          "averageItemLevel": null,
          "bossPercentage": null,
          "completeRaid": false,
          "difficulty": null,
          "encounterID": 0,
          "endTime": 4177668,
          "fightPercentage": null,
          "id": 35,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": null,
          "lastPhase": 0,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "Corrosive Digester",
          "rating": null,
          "size": null,
          "startTime": 4169650,
          "wipeCalledTime": null,
        },
        Object {
          "averageItemLevel": 465.4333190917969,
          "bossPercentage": 0,
          "completeRaid": false,
          "difficulty": 4,
          "encounterID": 2343,
          "endTime": 4519686,
          "fightPercentage": 0,
          "id": 36,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": true,
          "lastPhase": 0,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "Drest'agath",
          "rating": null,
          "size": 30,
          "startTime": 4350264,
          "wipeCalledTime": null,
        },
        Object {
          "averageItemLevel": null,
          "bossPercentage": null,
          "completeRaid": false,
          "difficulty": null,
          "encounterID": 0,
          "endTime": 4655283,
          "fightPercentage": null,
          "id": 37,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": null,
          "lastPhase": 0,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "Eye of the Depths",
          "rating": null,
          "size": null,
          "startTime": 4541872,
          "wipeCalledTime": null,
        },
        Object {
          "averageItemLevel": null,
          "bossPercentage": null,
          "completeRaid": false,
          "difficulty": null,
          "encounterID": 0,
          "endTime": 4716416,
          "fightPercentage": null,
          "id": 38,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": null,
          "lastPhase": 0,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "Leeching Parasite",
          "rating": null,
          "size": null,
          "startTime": 4682259,
          "wipeCalledTime": null,
        },
        Object {
          "averageItemLevel": 465.29998779296875,
          "bossPercentage": 0,
          "completeRaid": false,
          "difficulty": 4,
          "encounterID": 2335,
          "endTime": 4948936,
          "fightPercentage": 0,
          "id": 39,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": true,
          "lastPhase": 3,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "Shad'har the Insatiable",
          "rating": null,
          "size": 30,
          "startTime": 4769559,
          "wipeCalledTime": null,
        },
        Object {
          "averageItemLevel": null,
          "bossPercentage": null,
          "completeRaid": false,
          "difficulty": null,
          "encounterID": 0,
          "endTime": 5035698,
          "fightPercentage": null,
          "id": 40,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": null,
          "lastPhase": 0,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "Corrosive Digester",
          "rating": null,
          "size": null,
          "startTime": 4982876,
          "wipeCalledTime": null,
        },
        Object {
          "averageItemLevel": null,
          "bossPercentage": null,
          "completeRaid": false,
          "difficulty": null,
          "encounterID": 0,
          "endTime": 5136587,
          "fightPercentage": null,
          "id": 41,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": null,
          "lastPhase": 0,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "Sanguimar",
          "rating": null,
          "size": null,
          "startTime": 5129287,
          "wipeCalledTime": null,
        },
        Object {
          "averageItemLevel": 465.6333312988281,
          "bossPercentage": 0,
          "completeRaid": false,
          "difficulty": 4,
          "encounterID": 2345,
          "endTime": 5470049,
          "fightPercentage": 0,
          "id": 42,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": true,
          "lastPhase": 0,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "Il'gynoth, Corruption Reborn",
          "rating": null,
          "size": 30,
          "startTime": 5215096,
          "wipeCalledTime": null,
        },
        Object {
          "averageItemLevel": null,
          "bossPercentage": null,
          "completeRaid": false,
          "difficulty": null,
          "encounterID": 0,
          "endTime": 5648525,
          "fightPercentage": null,
          "id": 43,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": null,
          "lastPhase": 0,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "Monstrous Behemoth",
          "rating": null,
          "size": null,
          "startTime": 5490612,
          "wipeCalledTime": null,
        },
        Object {
          "averageItemLevel": null,
          "bossPercentage": null,
          "completeRaid": false,
          "difficulty": null,
          "encounterID": 0,
          "endTime": 5767141,
          "fightPercentage": null,
          "id": 44,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": null,
          "lastPhase": 0,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "Amalgamation of Flesh",
          "rating": null,
          "size": null,
          "startTime": 5678746,
          "wipeCalledTime": null,
        },
        Object {
          "averageItemLevel": 465.73333740234375,
          "bossPercentage": 0,
          "completeRaid": false,
          "difficulty": 4,
          "encounterID": 2337,
          "endTime": 6285751,
          "fightPercentage": 0,
          "id": 45,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": true,
          "lastPhase": 3,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "Carapace of N'Zoth",
          "rating": null,
          "size": 30,
          "startTime": 5889488,
          "wipeCalledTime": null,
        },
        Object {
          "averageItemLevel": 465.9666748046875,
          "bossPercentage": 2.21,
          "completeRaid": false,
          "difficulty": 4,
          "encounterID": 2344,
          "endTime": 6983364,
          "fightPercentage": 2.21,
          "id": 46,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": true,
          "lastPhase": 2,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "N'Zoth the Corruptor",
          "rating": null,
          "size": 30,
          "startTime": 6431775,
          "wipeCalledTime": null,
        },
        Object {
          "averageItemLevel": null,
          "bossPercentage": null,
          "completeRaid": false,
          "difficulty": null,
          "encounterID": 0,
          "endTime": 6987898,
          "fightPercentage": null,
          "id": 47,
          "keystoneAffixes": null,
          "keystoneBonus": null,
          "keystoneLevel": null,
          "keystoneTime": null,
          "kill": null,
          "lastPhase": 0,
          "lastPhaseIsIntermission": false,
          "layer": null,
          "name": "N'Zoth the Corruptor",
          "rating": null,
          "size": null,
          "startTime": 6983395,
          "wipeCalledTime": null,
        },
      ],
      "owner": Object {
        "id": 381152,
        "name": "Fluclox",
      },
      "revision": 49,
      "segments": 49,
      "startTime": 1602163523065,
      "title": "Heroic FFA Boost",
      "visibility": "public",
      "zone": Object {
        "id": 24,
        "name": "Ny'alotha",
      },
    },
  ],
  "from": 1,
  "has_more_pages": true,
  "last_page": 26,
  "per_page": 1,
  "to": 1,
  "total": 26,
}
`;
